core_backend/app/dashboard/models.py

```diff
@@ -0,0 +1,256 @@
+from datetime import date
+from typing import Dict, Union
+
+from sqlalchemy import func
+from sqlalchemy.ext.asyncio import AsyncSession
+from sqlalchemy.future import select
+
+from ..question_answer.models import (
+    ContentFeedbackDB,
+    QueryDB,
+    ResponseFeedbackDB,
+)
+from ..urgency_detection.models import UrgencyResponseDB
+from .schemas import (
+    ContentFeedbackStats,
+    QueryStats,
+    ResponseFeedbackStats,
+    StatsCards,
+    UrgencyStats,
+)
+
+
+async def get_stats_cards(
+    user_id: int, asession: AsyncSession, start_date: date, end_date: date
+) -> StatsCards:
+    """
+    Retrieve statistics for question answering and upvotes
+    """
+
+    query_stats = await get_query_count_stats(user_id, asession, start_date, end_date)
+    response_feedback_stats = await get_response_feedback_stats(
+        user_id, asession, start_date, end_date
+    )
+    content_feedback_stats = await get_content_feedback_stats(
+        user_id, asession, start_date, end_date
+    )
+    urgency_stats = await get_urgency_stats(user_id, asession, start_date, end_date)
+
+    return StatsCards(
+        query_stats=query_stats,
+        response_feedback_stats=response_feedback_stats,
+        content_feedback_stats=content_feedback_stats,
+        urgency_stats=urgency_stats,
+    )
+
+
+async def get_query_count_stats(
+    user_id: int, asession: AsyncSession, start_date: date, end_date: date
+) -> QueryStats:
+    """
+    Retrieve statistics for question answering
+    """
```

Comment:
Would it be possible to get that in a single query?

Reply:
Actually, since we will need a line chart as well, maybe just have an endpoint you will use for both?

Reply:
I'll refactor as I add more charts. Though an extra Db call is not that much of an overhead since it's all on the server side. And it makes for nice modularised code

Reply:
Actually, good point. We probably can combine it here since it's all inside one function. Let me do that

Reply:
Fixed